src/praisonai/praisonai/agents_generator.py

```
@@ -411,65 +411,13 @@ def _prepare_for_run(self, config):
-            self.logger.debug("tools.py exists in the root directory. Loading tools.py and skipping tools folder.")
-        elif tools_dir_path.is_dir():
-            tools_dict.update(self.tool_resolver.get_local_tool_classes_from_dir(tools_dir_path))
-            if tools_dict:
-                self.logger.debug("tools folder exists in the root directory")
-        
+        tools_dict = self.tool_resolver.resolve_all_from_yaml(config)
+        for tool_class in self.tools:
+            if isinstance(tool_class, type):
+                try:
+                    tools_dict[tool_class.__name__] = tool_class()
+                except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)